src/pages/index.js

```diff
@@ -96,142 +103,152 @@ popUpWithImage.setEventListeners();
-.catch((err) => {
-  console.error(err);
-  reject(err);
-}) 
-.finally(() => {
-  popUpAddItem.close();
-})
+    api
+      .createCard({ name: title, link })
+      .then((card) => {
+        section.addItem(createCard(card));
+        //console.log("CARD CHECK! please please work T_T.", card);
+        resolve();
+        popUpAddItem.close();
+      })
+      .catch((err) => {
+        console.error(err);
+        reject(err);
+      })
   });
-
 }
 
-
 function handleImageClick(data) {
   popUpWithImage.open(data);
 }
 
 function handleTrashButtonClick(item) {
-  console.log("enter trash index.js")
-  popUpConfirm.setLoading(true);
-  popUpConfirm.setSubmitCall(() => { 
-    api.deleteCard(item.getId())
-    .then(() => {
-      //const cardElement = document.querySelector(`[data-card-id="${cardId}"]`);
-      //if (cardElement){
-        popUpConfirm.close();
+  console.log("enter trash index.js");
+  popUpConfirm.setDeleting(true);
+// popUpConfirm.setDeleting(false); 
+  popUpConfirm.setSubmitCall(() => {
+    api
+      .deleteCard(item.getId())
+      .then(() => {
+        //const cardElement = document.querySelector(`[data-card-id="${cardId}"]`);
+        //if (cardElement){
+      
         item.removeCard();
-     // }
-    //popUpConfirm.setLoading(false);
-
-    })
-    .catch((err) => {
-      console.error("Error:", err);
-     //popUpConfirm.setLoading(false);
-    })
-   .finally(() => {
-    popUpConfirm.setLoading(false);
-   })
-  })
-  popUpConfirm.close();
+        popUpConfirm.close();
+      
+        // }
+        //popUpConfirm.setLoading(false);
+      })
+      .catch((err) => {
+        console.error("Error:", err);
+        //popUpConfirm.setLoading(false);
+      })
+      .finally (() =>   popUpConfirm.setDeleting(false)); 
+  });
+  //popUpConfirm.close();
+  
   popUpConfirm.open();
- console.log("exit trash index.js")
-}  
+  //console.log("exit trash index.js");
+}
 
 function handleHeartButton(item) {
+  const newIsLikedStatus = !item.isLiked;
+
   //item.isLiked = !item.isLiked;
 
-  if (item.isLiked) {
-    api.unlikeCard(item.getId())
-      .then(() => {
-        item.setLikeStatus(item.isLiked = false);
+  if (newIsLikedStatus) {
+    api
+      .likeCard(item.getId())
+      .then((respond) => {
+        console.log("unlike me", respond);
+        item.setLikeStatus(respond.isLiked);
+        //item.getCardElement().classList.remove("card__like-button_active")
       })
       .catch((err) => {
         console.error("Error:", err);
       });
   } else {
-    api.likeCard(item.getId())
-      .then(() => {
-        item.setLikeStatus(item.isLiked = true);
+    api
+      .unlikeCard(item.getId())
+      .then((respond) => {
+        console.log(respond)
+        item.setLikeStatus(respond.isLiked);
+       // item.getCardElement().classList.add("card__like-button_active")
       })
       .catch((err) => {
         console.error("Error:", err);
       });
   }
+  item.isLiked = newIsLikedStatus;
```

Comment:
`  item.isLiked = newIsLikedStatus;
`

Please, delete it

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.